vivisect/__init__.py

```diff
@@ -1029,7 +1029,7 @@ def detectString(self, va):
                             # technically the start of the full string, but the binary does
                             # some optimizations and just ref's inside the full string to save 
                             # some space
-                            return count + loc[L_SIZE]
```

Comment:
Hmm, let's walk through this? Because these pair of returns (on line 1032 and 1033) are trying  to cover two cases:
* 1032 is supposed to handle the case where we discover the inner string first, and then detectString is called on a VA before that one. Since count is how far we've gotten into scanning a byte range, we use it to determine how big the outer string should be and add that to the length of what we already know.
* 1033 handles vice versa, where we make the longer string first, and then get a ref inside the longer string, so the length returned needs to be shorter. The return could be cleaned up a lot.

So let's workshop this a bit? I'm not fully confident we're covering both cases? But I admit, it's been a long time since I've revisited this particular section of code.

Reply:
@williballenthin bump :)  i know you have nothing else to do in life. ;)
thanks for your help.

Reply:
@williballenthin is this still a valuable thing?  @rakuy0 would like to walk through how to solve this in the best fashion, but we've heard nothing for months.  thanks :)

Reply:
i'll provide a test case and another explanation shortly. thanks for your patience!